util.c

```diff
@@ -28,6 +30,13 @@ struct safe_buffer * new_safe_buffer(void)
 	return sb;
 }
 
+struct safe_buffer * clone_safe_buffer(struct safe_buffer const *src)
```

Comment:
It looks like this function is tested, but I don't see it being called anywhere else.  Is it being used?  If not, please remove it.